[style] 마이 피드 뷰 레이아웃 rough ver.

minseungseon opened the pull request:
## 작업 내용 
- [ ] rough 한 레이아웃만 잡았음
- [ ] collection view 추가, nib 사용해보았음

## 관계된 이슈, PR : 
resolves #61 

## 스크린샷
|SE2|12mini|12pro|
|:------:|:------:|:------:|
|<img src="https://user-images.githubusercontent.com/37579661/103976691-ca04a100-51ba-11eb-8437-d5f5a9dfb403.png" width=300> |  <img src="https://user-images.githubusercontent.com/37579661/103976713-dd177100-51ba-11eb-8afd-5879324f08d9.png" width=300> |<img src="https://user-images.githubusercontent.com/37579661/103976740-ec96ba00-51ba-11eb-9af5-1b140ba4fdaa.png" width=300> |

## 레퍼런스
https://www.youtube.com/watch?v=eWGu3hcL3ww&pbjreload=101

## 수정할 부분  
- 매개변수화 할 부분이 아직 남아있어서 추가 커밋으로 수정할 예정  
- collection view 에서 top 1, 좌우 0.5 , 이런식으로 inset 을 설정했는데 device 에 따라 달라지게 해줘야 하는 것으로 보임.. 미세한 차이가 보임.. 이 부분을 좀더 알아보고 수정해보겠음.

minseungseon committed:
[fix] DailyDiaryVC placeholder 문제 해결, 그 외 textView 관련 문제도 해결
body:
1. placeholder 가 다른데에 focus 갔다가 다시 돌아오면 Placeholder 가 사라지지 않는문제가 있어서 해결함
2. 직접 device test 를 했을 때 일기, 짧은 독서 뷰에서 글자를 쓴 후 화면 누르면 키보드 내려가는 기능이 없어서 추가함, @xwoud 의 코드 일부분 사용함.
3. [layout] 모든 뷰의 headerLabel 의 색깔이 지정 안되어 있었음. 설정해주었음.

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into fix/placeholder

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/PictureUpload

minseungseon committed:
[style] 기상 인증 뷰 layout 완료
body:
dquote> 1. 이름과 같은 데이터전달이 될 매개변수는 기존에 @xwoud 님이 사용했던 변수 이름 그대로 userNick 으로 설정했음.
dquote> 2. uploadedPicture 는 UIImageView이며, 일단 test_img 파일을 assets 에 추가해서 연결해두었음. 실제 서버 연결에서는 회의 때 언급했던 kingfisher 의 image cashing 을 통해 이 부분에 데이터 전달을 하면
될 것으로 보임.
3. 22번째의 숫자도 매개변수화 해야할 것으로 보임.

minseungseon committed:
[refac] 숫자 변수화, letterSpacing 조정
body:
1. ~번째 의미있는 아침에서 숫자를 변수화 해주었음.
2. bodyUpperLabel 의 lineSpacing 을 조절하였음.. Zeplin 에 있는 것과 다른 숫자 개념인 것 같음. 일단 외관상 가장 비슷한 정도인 6 으로 line spacing 을 주었는데 이 부분에 대해서 이 extension 을 만든 본인이 검토를 해보도록 하겠습니다.

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/PictureUpload

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/PictureUpload

minseungseon committed:
[refac] 리뷰 피드백 반영
1. convention 수정
2. uploadedImage 데이터 전달 방식에 따라 변수화

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/myPage

minseungseon committed:
[style] 마이 피드 뷰 레이아
body:
1. rough 한 레이아웃만 잡았음
2. collection view 추가, nib 사용해보았음

pk33n commented on meaning/meaning/Screen/MyPage/ViewController/MyPageFeedVC.swift:
```diff
@@ -0,0 +1,155 @@
+    
+    func setLayout() {
+        
+        let initial: String = String(self.userNick.prefix(1))
```
오 이게 더 간단해보이네요

pk33n reviewed (approved):
민뜽 넘 고생했어요 ,, !!!! 진짜 잘 한다

xwoud commented on meaning/meaning/Screen/MyPage/ViewController/MyPageFeedVC.swift:
```diff
@@ -0,0 +1,155 @@
+        self.userWakeTimeView.setRounded(radius: 13.5)
+        self.userWakeTimeView.backgroundColor = .meaningWhite
+        
+        self.userWakeTimeLabel.text = "매일 오전 5시 기상"
```
5시는 변수로 만드는거 추천해요!

minseungseon replied:
네넵 변수화는 일단 지금 하고있고 혹시나 다른 부분 수정할 게 있을 지 확인받으려고 임시 차원에서 올린거였습니다!!!!!

xwoud commented on meaning/meaning/Screen/MyPage/ViewController/MyPageFeedVC.swift:
```diff
@@ -0,0 +1,155 @@
+        self.userWakeTimeLabel.textAlignment = .center
+        
+        self.feedHeaderView.backgroundColor = .meaningLightblue
+        self.feedHeaderLabel.text = "오늘은 365일 중에 30번째 의미있는 아침입니다."
```
30 도 변수로 만드는것을 추천합니다! 아니면 collectionview 개체의 count로 줘도 되겠군요!

minseungseon replied:
헐 좋은 아이디어 감사합니다ㅠㅠ

xwoud commented on meaning/meaning/Screen/MyPage/ViewController/MyPageFeedVC.swift:
```diff
@@ -0,0 +1,155 @@
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        collectionView.deselectItem(at: indexPath, animated: true)
+        
+        print("tapped")
```
print문이나 쓸모없는 코드는 지우고 올려주세요 'ㅅ'

minseungseon replied:
헉 넵

xwoud commented on meaning/meaning/Screen/MyPage/cell/FeedCollectionViewCell.swift:
```diff
@@ -0,0 +1,28 @@
+
+    @IBOutlet var feedImage: UIImageView!
+    
+    static let identifier = "FeedCollectionViewCell"
```
코드 위치를 보면 상수변수 선언이 ```IBOutlet``` 보다 위에 있어야한다는것>_< 우리의 약속!

xwoud commented on meaning/meaning/Screen/MyPage/ViewController/MyPageFeedVC.swift:
```diff
@@ -0,0 +1,155 @@
+        super.viewDidLoad()
+        setLayout()
+        
+        let layout = UICollectionViewFlowLayout()
+//        layout.itemSize = CGSize(width: 120, height: 120)
+        feedCollectionView.collectionViewLayout = layout
```
이 두 줄은 어떤 용도로 넣어준것인가요?

minseungseon replied:
flowLayout 아래 extension 을 적용해주려는 용도입니다! 저 주석처리 한 것은 삭제해서 다시 커밋하겠습니다!

minseungseon committed:
[refac] FeedImage model 추가, 연동 완료
body:
1. FeedImage struct 만들어두었음
2. @xwoud 의 피드백에 따라 ~번째 아침에서 숫자를 feedImageList 의 count로 설정하였음.

minseungseon committed:
Merge branch 'develop' of https://github.com/nneaning/meaning_iOS into 2nd/myPage

minseungseon committed:
[fix] collection view의 scroll view inset 조정
dquote> body:
dquote> 1. iphone SE 외의 큰 device 에서 collection view 아래 부분 safe area 가 비어있게 스크롤이 되는 문제 해결
2. UIScrollViewContentInsetAdjustment .never 로 설정함
3. 간단한 주석 추가함

minseungseon committed:
[refac] collectino view 부분 코드 정리

minseungseon committed:
[refac] random 숫자0,1에 따라 profile 색깔 변하도록 적용함
body:
1. random number 적용'
2. subview 에서는 profile view 원으로 만들고, 색깔 적용
3. 아래 setLayout 에서는 textcolor 적용
+ subview 에서 textlabel 색깔을 한꺼번에 바꾸려고 하니까 적용이 안되었었음.. 왜그런지에 대해서 궁금함.
4. FeedCollectionViewCell.swift 주석처리 함

minseungseon committed:
[upload] 매일 오전 ~시 ~분 기상 UIView 를 위한 새로운 Extension 추가
body:
1. 이름의 새로운 extension file added.
2. 상세 설명 읽기 바람
3. 마이 피드 뷰에도 새로운 extension 적용 완료

pk33n commented on meaning/meaning/Global/Extension/WakeUpTimeView+Extension.swift:
```diff
@@ -0,0 +1,47 @@
+
+// 사용법: self.userWakeTimeLabel.text = dateConverter(dateData: "00:00:00", originalConstraint: timeViewWidth, convertedWidth: 150)
+
+extension UIViewController {
```
새로운 extension 까지 만들다니  ,,, 갓민승 ,,
